Src/PerformanceCollector/Shared/Implementation/StandardPerformanceCollector/StandardPerformanceCollector.cs

```diff
@@ -0,0 +1,342 @@
+        private void RefreshCounter(PerformanceCounterData pcd)
+        {
+            string dummy;
+
+            bool usesInstanceNamePlaceholder;
+            var pc = this.CreateAndValidateCounter(
+                pcd.OriginalString,
+                out usesInstanceNamePlaceholder,
+                out dummy);
+
+            try
+            {
+                this.RefreshPerformanceCounter(pcd);
+
+                PerformanceCollectorEventSource.Log.CounterRegisteredEvent(
+                        PerformanceCounterUtility.FormatPerformanceCounter(pc));
+            }
+            catch (InvalidOperationException e)
+            {
+                PerformanceCollectorEventSource.Log.CounterRegistrationFailedEvent(
+                    e.Message,
+                    PerformanceCounterUtility.FormatPerformanceCounter(pc));
+            }
+        }
+
+        /// <summary>
+        /// Registers the counter to the existing list of counters.
+        /// </summary>
+        /// <param name="originalString">Counter original string.</param>
+        /// <param name="reportAs">Counter report as.</param>
+        /// <param name="pc">Performance counter.</param>
+        /// <param name="isCustomCounter">Boolean to check if it is a custom counter.</param>
+        /// <param name="usesInstanceNamePlaceholder">Uses Instance Name Place holder boolean.</param>
+        /// <param name="error">Error message.</param>
+        private void RegisterCounter(
+            string originalString,
+            string reportAs,
+            PerformanceCounter pc,
+            bool isCustomCounter,
+            bool usesInstanceNamePlaceholder,
+            out string error)
+        {
+            error = null;
+
+            try
+            {
+                this.RegisterPerformanceCounter(
+                    originalString,
+                    reportAs,
+                    pc.CategoryName,
+                    pc.CounterName,
+                    pc.InstanceName,
+                    usesInstanceNamePlaceholder,
+                    isCustomCounter);
+
+                PerformanceCollectorEventSource.Log.CounterRegisteredEvent(
+                    PerformanceCounterUtility.FormatPerformanceCounter(pc));
+            }
+            catch (InvalidOperationException e)
+            {
+                PerformanceCollectorEventSource.Log.CounterRegistrationFailedEvent(
+                    e.Message,
+                    PerformanceCounterUtility.FormatPerformanceCounter(pc));
+                error = e.Message;
+            }
+        }
+
+        /// <summary>
+        /// Generates Unique key for a specific performance counter data using the category name, counter name and instance name.
+        /// </summary>
+        /// <param name="pcd">Target performance counter data.</param>
+        /// <returns>Unique key for the specific counter data.</returns>
+        private string GenerateKeyForPerformanceCounter(PerformanceCounterData pcd)
```

Comment:
Removed the dictionary, and retained the counter objects in the same list, and what we expose out is restricted to the common data object.